src/oas3/transformers/schema.ts

```diff
@@ -0,0 +1,152 @@
+import type { JSONSchema4, JSONSchema4TypeName } from 'json-schema';
```

Comment:
Interesting that this converter is in the `oas3` transformers directory. Does this process not apply to `oas2`?

IF this is indeed rather generic "turn json schema into openapi compliant draft 4" functionality, maybe it belongs in the `oas` directory?

Reply:
> Interesting that this converter is in the oas3 transformers directory. Does this process not apply to oas2?

perhaps it's because oas2 doesn't support draft 4? not sure.

Reply:
OAS2 and OAS3 are both based on draft 4 (technically oas3 is draft 5 but they’re basically the same thing in any way that matters).

Reply:
ah, sorry, I wasn't clear.
What I meant was that the oas2 json schema flavor has different keywords, that the library doesn't recognize.

I might be wrong, though, and perhaps the library yields decent results for OAS2.